File: Practica1/Ejercicio4/src/practica1ejercicio4/FiltroCorreo.java
package practica1ejercicio4;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class FiltroCorreo implements Filtro{

    @Override
    public boolean ejecutar(Correo correo) {
        String regex = "^[^@]+@(gmail|hotmail)\\.com$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(correo.email);
        return matcher.matches();
    }
    
}

File: Practica2/ejercicio_grupal1/lib/FiltroCorreo.dart
import 'Filtro.dart';
import 'Correo.dart';

class FiltroCorreo implements Filtro {
  @override
  MapEntry<bool,String> ejecutar(Correo correo) {
    MapEntry<bool, String> output;
    final regex = RegExp(r'^[a-zA-Z0-9._%+-]+@(?:gmail|hotmail)\.com$');

    if (regex.hasMatch(correo.email)) { // En caso de que el correo haga match
      output = MapEntry(true, "");
    }
    else{
      output = MapEntry(false, "Correo debe de tener la estructura: "
          "(letra o número)@gmail|hotmail.com");
    }
    return output;
  }
}